[tf-psa-crypto] tests: pk: add a common function to create a PSA key out of predefined keys

valeriosetti opened the pull request:
## Description

Resolves #598 
Depends on https://github.com/Mbed-TLS/mbedtls-framework/pull/257

## PR checklist

- [x] **changelog** not required because: only test changes
- [x] **framework PR** provided https://github.com/Mbed-TLS/mbedtls-framework/pull/257
- [x] **mbedtls development PR** not required because: no change there
- [x] **mbedtls 3.6 PR** not required because: no backport
- **tests**  provided

valeriosetti committed:
tests: pk: move helpers to create a key from predefined data to framework

Make a common helper between test_suite_pk and test_suite_pkwrite and move
it to the framework.

Signed-off-by: Valerio Setti <vsetti@baylibre.com>

valeriosetti committed:
tests: pk: move helpers to populate PK context to framework

Signed-off-by: Valerio Setti <vsetti@baylibre.com>

valeriosetti committed:
framework: update reference

Signed-off-by: Valerio Setti <vsetti@baylibre.com>

valeriosetti committed:
tests: pk: minor changes to align to changes in PK helpers

PK helpers were sligthly modified in the framework and this commit aligns
to those changes.

Signed-off-by: Valerio Setti <vsetti@baylibre.com>

gilles-peskine-arm commented on tests/suites/test_suite_pk.function:
```diff
@@ -239,8 +206,10 @@ static int pk_setup(mbedtls_pk_context *pk, mbedtls_pk_type_t pk_type, int curve
 #if defined(PSA_WANT_KEY_TYPE_RSA_PUBLIC_KEY)
-        TEST_EQUAL(get_predefined_key_data(0, curve_or_keybits, &key_data, &key_data_len,
-                                           &pub_key_data, &pub_key_data_len), 0);
+        mbedtls_pk_helpers_get_predefined_key_data(PSA_KEY_TYPE_RSA_KEY_PAIR, curve_or_keybits,
```
Here and elsewhere: if the function fails, the test case is marked as fail, but the execution will continue until something else notices a problem. It would be better to return early (or goto exit if there's something to clean up). That's not critical, since it only matters if the test case has already failed. But it would help the next person who's trying to understand how the test is failing.

valeriosetti replied:
Yeah, sorry. This dates back to the original implementation of `mbedtls_pk_helpers_get_predefined_key_data` which didn't returned any value. I will update this immediately

gilles-peskine-arm reviewed (dismissed):
LGTM, although the behavior on failure is fragile and I'd prefer to properly stop on failure.

valeriosetti committed:
tests: pk: fix failure handling in 'pk_setup'

If 'mbedtls_pk_helpers_get_predefined_key_data' fails then the test is
already marked as failed so:
- there is no need to use TEST_EQUAL() again;
- execution should stop immediately and jump to 'exit' label.

While at it also use PSA_ASSERT() instead of TEST_EQUAL() when calling
PSA functions.

Signed-off-by: Valerio Setti <vsetti@baylibre.com>

gilles-peskine-arm commented on tests/suites/test_suite_pk.function:
```diff
@@ -377,30 +360,13 @@ static psa_status_t pk_psa_setup(psa_key_type_t type, size_t bits,
-        TEST_FAIL("EC keys are not supported");
-#endif /* PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY */
+    if (PSA_KEY_TYPE_IS_RSA(type) || PSA_KEY_TYPE_IS_ECC(type)) {
+        mbedtls_pk_helpers_get_predefined_key_data(type, bits, &key_data, &key_data_size);
```
This is still missing the value return check. Otherwise LGTM

valeriosetti committed:
tests: pk: fix error handling in pk_psa_setup()

Signed-off-by: Valerio Setti <vsetti@baylibre.com>

gilles-peskine-arm reviewed (approved):
LGTM
